File: Level1/Ponketmon/Solution.js
function solution(nums) {
    let maxi = Math.floor(nums.length / 2)
    let ans = []
    for (let i = 0; i < nums.length; i++) {
        if (!ans.includes(nums[i])) {
            ans.push(nums[i])
        }
    }
    return Math.min(ans.length, maxi);
}

File: Level1/Ponketmon/Solution.java
package Ponketmon;
import java.util.*;
public class Solution {
    public static int solution(int[] nums) {
        int answer = 1;
        Arrays.sort(nums);
        int n = nums[0];
        for (int num : nums) {
            if (n != num) {
                answer++;
            }
            n = num;
        }
        return Math.min(answer, nums.length / 2);
    }
}